WordPress/src/main/java/org/wordpress/android/ui/comments/CommentDetailFragment.java

```diff
@@ -687,33 +690,14 @@ private void moderateComment(
         // Fire the appropriate listener if we have one
         if (note != null && mOnNoteCommentActionListener != null) {
             mOnNoteCommentActionListener.onModerateCommentForNote(note, newStatus);
-            dispatchModerationAction(site, comment, newStatus);
+            mViewModel.dispatchModerationAction(site, comment, newStatus);
```

Comment:
When the comment is from a notification, we can  `dispatchModerationAction` so it can update the comment status without leaving the screen.

P.S. When DELETE a comment, it will finish the screen after the deletion.